std-bits/table/src/main/java/org/enso/table/data/column/builder/DoubleBuilder.java

```diff
@@ -86,24 +87,36 @@ public void append(Object o) {
   public void appendBulkStorage(Storage<?> storage) {
     if (Objects.equals(storage.getType(), FloatType.FLOAT_64)) {
       if (storage instanceof DoubleStorage doubleStorage) {
-        int n = doubleStorage.size();
+        int n = (int) doubleStorage.getSize();
         ensureFreeSpaceFor(n);
         System.arraycopy(doubleStorage.getRawData(), 0, data, currentSize, n);
         BitSets.copy(doubleStorage.getIsNothingMap(), isNothing, currentSize, n);
         currentSize += n;
+      } else if (storage instanceof ColumnDoubleStorage doubleStorage) {
+        long n = doubleStorage.getSize();
+        for (long i = 0; i < n; i++) {
+          if (storage.isNothing(i)) {
+            appendNulls(1);
+          } else {
+            appendDouble(doubleStorage.getItemAsDouble(i));
+          }
+        }
```

Comment:
About these: I'm a little bit slightly worried that because currently all (most?) storages are the 'base' e.g. `DoubleStorage`, this more 'abstract' variant will rarely run and so is not covered that much by tests.

This is a recurring problem that there are often more edge cases in the implementation of these low-level operations than seems feasible to have high-level tests (especially also ones shared with DB which are often slower).

Not for this PR but perhaps we need to add some 'unit' tests to `std-table` that test the internals more, to reinforce our correctness here?

Reply:
No `DoubleStorage` is a `ColumnDoubleStorage` so this will run.
All the typed storages are their specific `ColumnStorage`.

Reply:
Agree on some tests on the builders to check we get the correct interfaces out makes sense.

Reply:
> No `DoubleStorage` is a `ColumnDoubleStorage` so this will run. All the typed storages are their specific `ColumnStorage`.

No, because above it is a more specific branch that has a specialization for `DoubleStorage` specifically. Which is good. But that means that specialization above runs, not this generic code. That was my point above.

Reply:
Ah ok - sorry yes.

Reply:
It felt sensible to add these in at this point but will revisit as pull operations out one by one.

Reply:
These will be used pretty quickly as `ConstantColumnDouble` would be a `ColumnDoubleStorage` for example.

Reply:
The number parser had a load of JUnit tests. @radeusgd would you be able to look at adding running JUnit tests at some point?